packages/xo-server/src/xapi-stats.js

```diff
@@ -77,17 +78,18 @@ const computeValues = (dataRow, legendIndex, transformValue = identity) =>
 const combineStats = (stats, path, combineValues) =>
   zipWith(...map(stats, path), (...values) => combineValues(values))
 
-// It browse the object in depth and initialise it's properties
+// It browse the object in depth and initialise it's properties if not exists
 // The targerPath can be a string or an array containing the depth
 // targetPath: [a, b, c] => a.b.c
-const getValuesFromDepth = (obj, targetPath) => {
+const getValuesFromDepth = (obj, targetPath, lastChildDefaultValue = []) => {
   if (typeof targetPath === 'string') {
-    return (obj[targetPath] = [])
+    return (obj[targetPath] = defined(obj[targetPath], lastChildDefaultValue))
```

Comment:
Use [the nullish coalescing operator](https://babeljs.io/docs/en/next/babel-plugin-proposal-nullish-coalescing-operator.html) instead of `defined`.